### PR TITLE
Adding Changes to mainTemplate #92

### PR DESCRIPTION
**Pull request by Arshpreet1002**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: lansa-vmss-windows-autoscale-sql-database/mainTemplate.json [hunk 1]
```diff
@@ -251,11 +251,11 @@
       },
       "minimumInstanceCount": {
          "type": "int",
-         "defaultValue": 1,
+         "defaultValue": 2,
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/mainTemplate.json]: This needs to still be 1

File: lansa-vmss-windows-autoscale-sql-database/mainTemplate.json [hunk 2]
```diff
@@ -251,11 +251,11 @@
...
          "metadata": {
             "description": "Minimum number of Virtual Machine instances (1 to 600)."
          },
-         "minValue": 1,
+         "minValue": 2,
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/mainTemplate.json]: Ditto

File: lansa-vmss-windows-autoscale-sql-database/mainTemplate.json [hunk 3]
```diff
@@ -653,6 +654,30 @@
...
+      "name": "[variables('frontEndNSGName')]",
+      "location": "[parameters('location')]",
+       "properties": {
+         "securityRules": [
+           {
+            "name": "rdp-rule",
+            "properties": {
+            "description": "Allow RDP",
+            "protocol": "Tcp",
+            "sourcePortRange": "*",
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/mainTemplate.json:667]: This SourcePort range must be the range of the NAT rules : natDBStartPort to natEndPort

File: lansa-vmss-windows-autoscale-sql-database/mainTemplate.json [hunk 4]
```diff
@@ -940,7 +925,7 @@
          "sku": {
             "name": "[parameters('virtualMachineSize')]",
             "tier": "Standard",
-            "capacity": "[parameters('minimumInstanceCount')]"
+            "capacity": 2
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/mainTemplate.json]: I said you needed to revert this change

File: lansa-vmss-windows-autoscale-sql-database/mainTemplate.json [hunk 5]
```diff
@@ -983,16 +968,14 @@
                                            "id": "[resourceId('Microsoft.Network/loadBalancers/backendAddressPools', variables('loadBalancerName'), variables('bePoolName'))]"
                                         }
                                      ],
-                                    "loadBalancerInboundNatPools": [
+                                     "loadBalancerInboundNatPools": [
                                        {
                                           "id": "[resourceId('Microsoft.Network/loadBalancers/inboundNatPools', variables('loadBalancerName'), variables('natPoolName'))]"
                                        }
-                                    ],
```
Comment (robe070) [inline lansa-vmss-windows-autoscale-sql-database/mainTemplate.json:990]: This should not be removed.

**Review by robe070 — CHANGES_REQUESTED** [review]

**Comment by robe070** [comment]
In order to test that the AG is working to each of the VMSS, its required to manually remove the VMSS from the AG and then use the url to test the other VMSS is working.
So here:
image

and then you can see that currently there is only 1 VMSS - because of the issue already identified in the template:
image

And remove 1 of the VMSS to test it, then add it back in and remove the other.

And I've just added the `testing-a` VMSS in manually to make sure thats possible - currently deploying

**Comment by robe070** [comment]
Please change dbVirtualMachineSize to have the same default as VirtualMachineSize. They are now equivalent VMs doing the same work. That is, Standard_B4ms

**Comment by robe070** [comment]
Add adding the VMSS failed with a 'try later' comment. Weird.

**Comment by robe070** [comment]
The pipeline is failing with a strange error. Have you deployed it yourself yet?

**Comment by Arshpreet1002** [comment]
I have checked it locally.It was running fine

**Comment by Arshpreet1002** [comment]
Ran Successfully.

**Comment by robe070** [comment]
so did you do the test in the AG to remove each of the VMSS in turn and test that the URL still worked?

**Comment by robe070** [comment]
> Please change dbVirtualMachineSize to have the same default as VirtualMachineSize. They are now equivalent VMs doing the same work. That is, Standard_B4ms

You still need to do this

**Comment by Arshpreet1002** [comment]
I have Tested By removing one of the VM's from AG and the URL is still Working.
And Chnaged the dbVitualMAchineSize to default .

**Pull request merged**